relay-general/src/store/normalize.rs

```diff
@@ -1001,6 +1002,78 @@ impl<'a> Processor for NormalizeProcessor<'a> {
         contexts.0.remove("reprocessing");
         Ok(())
     }
+
+    fn process_event_logger(
+        &mut self,
+        event_logger: &mut EventLogger,
+        meta: &mut Meta,
+        state: &ProcessingState,
+    ) -> ProcessingResult {
+        event_logger.process_child_values(self, state)?;
+
+        if event_logger.is_empty() {
+            return Ok(());
+        }
+
+        let trimmed_logger = event_logger.as_str().trim();
+        let logger_len = bytecount::num_chars(trimmed_logger.as_bytes());
+
+        if trimmed_logger.is_empty() {
+            return Err(ProcessingAction::DeleteValueHard);
+        } else if logger_len > MaxChars::Logger.limit() {
+            meta.add_error(Error::new(ErrorKind::ValueTooLong));
+            meta.set_original_length(Some(logger_len));
+
+            let new_logger = shorten_logger(trimmed_logger);
+            event_logger.set_logger(new_logger);
+        } else if trimmed_logger != event_logger.as_str() {
+            event_logger.set_logger(trimmed_logger.to_owned());
+        }
+
+        Ok(())
+    }
+}
+
+static LOGGER_DELIMITER_PATTERN: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"\.").expect("Invalid logger separator Regex"));
+static LOGGER_TRIMMED_PREFIX: &str = "*";
+
+fn shorten_logger(logger: &str) -> String {
+    let mut tokens = LOGGER_DELIMITER_PATTERN.split(logger).collect_vec();
```

Comment:
Using regex here seems like overkill to me.
```suggestion
    let mut tokens = logger.split(LOGGER_DELIMITER_CHAR).collect_vec();
```